lightning-invoice/src/utils.rs

```diff
@@ -221,6 +186,74 @@ where
 	}
 }
 
+/// Filters the `channels` for an invoice, and returns the corresponding `RouteHint`s to include
+/// in the invoice.
+///
+/// The filtering is based on the following criteria:
+/// * Only one channel per counterparty node
+/// * Always select the channel with the highest inbound capacity per counterparty node
+/// * Filter out channels with a lower inbound capacity than `min_inbound_capacity_msat`, if any
+/// channel with a higher or equal inbound capacity than `min_inbound_capacity_msat` exists
+/// * If any public channel exists, the returned `RouteHint`s will be empty, and the sender will
+/// need to find the path by looking at the public channels instead
+fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>) -> Vec<RouteHint>{
+	let mut filtered_channels: HashMap<PublicKey, ChannelDetails> = HashMap::new();
```

Comment:
I wasn't able to make this take a borrowed `&ChannelDetails`, as the borrow doesn't live outside of the channel loop context. Is that ok?

Reply:
Instead of borrowing `channel` directly within the loop, you can change the loop to iterate over references using:

```rust
for channel in channels.iter() {
```

Otherwise, `into_iter` is implicitly used, which consumes `channels` and yields values instead of references.

Reply:
Oh nice, thanks for clarifying!